modular_zubbers/modules/primitive_cooking/code/butterchurn.dm

```diff
@@ -0,0 +1,102 @@
+
+/obj/structure/butterchurn/atom_deconstruct(disassembled)
+	var/obj/item/stack/sheet/mineral/wood/plank = new (drop_location())
+	plank.amount = 8
+	plank.update_appearance(UPDATE_ICON)
+	transfer_fingerprints_to(plank)
+	return ..()
+
+/obj/structure/butterchurn/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+	balloon_alert_to_viewers("disassembling...")
+	if(!do_after(user, 2 SECONDS, src))
+		return
+	deconstruct(TRUE)
+
+/obj/structure/butterchurn/click_ctrl_shift(mob/user)
+	set_anchored(!anchored)
+	balloon_alert(user, "[anchored ? "secured" : "unsecured"]")
+
+/obj/structure/butterchurn/attack_hand(mob/living/carbon/user, list/modifiers)
+	if(!can_interact(user))
+		return
+
+	if(busy)
+		to_chat(user, span_warning("Someone's already using \the [src]!"))
+		return
+
+	if(user.getStaminaLoss() > CHURN_STAMINA_MINIMUM)
+		balloon_alert(user, "too tired")
+		return
+
+	if(!reagents.has_reagent(/datum/reagent/consumable/milk) && !reagents.has_reagent(/datum/reagent/consumable/cream))
+		balloon_alert(user, "nothing to churn")
+		return
+
+	if(!reagents.has_reagent(/datum/reagent/consumable/milk) && reagents.has_reagent(/datum/reagent/consumable/cream) && reagents.get_reagent_amount(/datum/reagent/consumable/cream) < CHURN_BUTTER_RATIO)
+		balloon_alert(user, "not enough cream")
+		return
+
+	balloon_alert_to_viewers("churning...")
+	playsound(src, 'sound/items/tools/crowbar_prying.ogg', 50, TRUE)
+	flick("churning", src)
+
+	var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/primitive, SKILL_SPEED_MODIFIER)
+	add_fingerprint(user)
+	busy = TRUE
+	if(do_after(user, 5 SECONDS * skill_modifier, src))
+		var/stamina_use = CHURN_STAMINA_USE
+		if(prob(user.mind.get_skill_modifier(/datum/skill/primitive, SKILL_PROBS_MODIFIER)))
+			stamina_use *= 0.5 //so it uses half the amount of stamina (25 instead of 50)
+
+		user.adjustStaminaLoss(stamina_use) // Prevents spamming it
+
+		busy = FALSE
+		user.visible_message(span_notice("[user] churns \the [src]."), span_notice("You finish churning \the [src]."))
+		var/cream_amt = reagents.get_reagent_amount(/datum/reagent/consumable/milk) / CHURN_CREAM_RATIO
+		var/cream_purity = reagents.get_reagent_purity(/datum/reagent/consumable/milk)
+		var/butter_amt = FLOOR(reagents.get_reagent_amount(/datum/reagent/consumable/cream) / CHURN_BUTTER_RATIO, 1)
+		var/butter_purity = reagents.get_reagent_purity(/datum/reagent/consumable/cream)
+		reagents.remove_reagent(/datum/reagent/consumable/cream, 10 * butter_amt)
+		for(var/i in 1 to butter_amt)
+			var/obj/item/food/butter/tasty_butter = new(drop_location())
+			tasty_butter.reagents.set_all_reagents_purity(butter_purity)
+		reagents.remove_reagent(/datum/reagent/consumable/milk, reagents.get_reagent_amount(/datum/reagent/consumable/milk))
+		reagents.add_reagent(/datum/reagent/consumable/cream, cream_amt, added_purity = cream_purity)
+	else
+		busy = FALSE
+		balloon_alert_to_viewers("stopped churning")
```

Comment:
This could be a bit neater if it was an if(!do_after) that did the busy false and balloon alert and then returned, and then everything after that can be not in the if

Reply:
In my defense, it was written at way the fuck too late at night/early in the morning. That's absolutely cleaner, implemented in latest commit!